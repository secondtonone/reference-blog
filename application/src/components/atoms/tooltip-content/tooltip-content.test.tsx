import renderer from 'react-test-renderer';
import TooltipContent from './tooltip-content';
import SvgIcon from '~/components/containers/svg-icon';

jest.mock('~/components/containers/svg-icon', () => jest.fn().mockReturnValue('svg-arrow'));

describe('Match snapshots TooltipContent', () => {
  it('empty', () => {
    const tree = renderer.create(<TooltipContent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with content', () => {
    const tree = renderer.create(<TooltipContent>content</TooltipContent>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with warning color', () => {
    const tree = renderer.create(<TooltipContent isAccent>content</TooltipContent>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with left arrow', () => {
    const tree = renderer.create(<TooltipContent isAccent arrowSide="left">content</TooltipContent>).toJSON();
    expect(tree).toMatchSnapshot();
    expect(SvgIcon).toHaveBeenCalledWith({
      code: 'arrow',
      mod: '-arrow-left',
    }, {});
  });
  it('with right arrow', () => {
    const tree = renderer.create(<TooltipContent isAccent arrowSide="right">content</TooltipContent>).toJSON();
    expect(tree).toMatchSnapshot();
    expect(SvgIcon).toHaveBeenCalledWith({
      code: 'arrow',
      mod: '-arrow-right',
    }, {});
  });
  it('with down arrow', () => {
    const tree = renderer.create(<TooltipContent isAccent arrowSide="down">content</TooltipContent>).toJSON();
    expect(tree).toMatchSnapshot();
    expect(SvgIcon).toHaveBeenCalledWith({
      code: 'arrow',
      mod: '-arrow-down',
    }, {});
  });
  it('with up arrow', () => {
    const tree = renderer.create(<TooltipContent isAccent arrowSide="up">content</TooltipContent>).toJSON();
    expect(tree).toMatchSnapshot();
    expect(SvgIcon).toHaveBeenCalledWith({
      code: 'arrow',
      mod: '-arrow-up',
    }, {});
  });
  it('default arrow is up', () => {
    const tree = renderer.create(<TooltipContent isAccent>content</TooltipContent>).toJSON();
    expect(tree).toMatchSnapshot();
    expect(SvgIcon).toHaveBeenCalledWith({
      code: 'arrow',
      mod: '-arrow-up',
    }, {});
  });
  it('with html', () => {
    const tree = renderer
      .create(
        <TooltipContent>
          <p>content</p>
          <ul>
            <li>1</li>
            <li>2</li>
            <li>3</li>
            <li>4</li>
            <li>5</li>
          </ul>
        </TooltipContent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
