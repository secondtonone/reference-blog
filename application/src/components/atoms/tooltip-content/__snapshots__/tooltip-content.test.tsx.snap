// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots TooltipContent default arrow is up 1`] = `
<div
  className="tooltip-content__TooltipContentStyled-sc-jyk063-0 csWTkH"
  data-test="tooltip-content"
>
  <span>
    content
  </span>
  svg-arrow
</div>
`;

exports[`Match snapshots TooltipContent empty 1`] = `
<div
  className="tooltip-content__TooltipContentStyled-sc-jyk063-0 csWTkH"
  data-test="tooltip-content"
>
  <span />
  svg-arrow
</div>
`;

exports[`Match snapshots TooltipContent with content 1`] = `
<div
  className="tooltip-content__TooltipContentStyled-sc-jyk063-0 csWTkH"
  data-test="tooltip-content"
>
  <span>
    content
  </span>
  svg-arrow
</div>
`;

exports[`Match snapshots TooltipContent with down arrow 1`] = `
<div
  className="tooltip-content__TooltipContentStyled-sc-jyk063-0 csWTkH"
  data-test="tooltip-content"
>
  <span>
    content
  </span>
  svg-arrow
</div>
`;

exports[`Match snapshots TooltipContent with html 1`] = `
<div
  className="tooltip-content__TooltipContentStyled-sc-jyk063-0 csWTkH"
  data-test="tooltip-content"
>
  <span>
    <p>
      content
    </p>
    <ul>
      <li>
        1
      </li>
      <li>
        2
      </li>
      <li>
        3
      </li>
      <li>
        4
      </li>
      <li>
        5
      </li>
    </ul>
  </span>
  svg-arrow
</div>
`;

exports[`Match snapshots TooltipContent with left arrow 1`] = `
<div
  className="tooltip-content__TooltipContentStyled-sc-jyk063-0 csWTkH"
  data-test="tooltip-content"
>
  <span>
    content
  </span>
  svg-arrow
</div>
`;

exports[`Match snapshots TooltipContent with right arrow 1`] = `
<div
  className="tooltip-content__TooltipContentStyled-sc-jyk063-0 csWTkH"
  data-test="tooltip-content"
>
  <span>
    content
  </span>
  svg-arrow
</div>
`;

exports[`Match snapshots TooltipContent with up arrow 1`] = `
<div
  className="tooltip-content__TooltipContentStyled-sc-jyk063-0 csWTkH"
  data-test="tooltip-content"
>
  <span>
    content
  </span>
  svg-arrow
</div>
`;

exports[`Match snapshots TooltipContent with warning color 1`] = `
<div
  className="tooltip-content__TooltipContentStyled-sc-jyk063-0 csWTkH"
  data-test="tooltip-content"
>
  <span>
    content
  </span>
  svg-arrow
</div>
`;
